test/tests/functional/pbs_equiv_classes.py

```diff
@@ -1773,145 +1753,124 @@ def test_multiple_equivalence_class_preemption(self):
 
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid2)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid4)
 
-        self.scheduler.log_match("Number of job equivalence classes: 1",
+        # Check equivalence classes =4
+        self.scheduler.log_match("Number of job equivalence classes: 4",
                                  max_attempts=10, starttime=self.t)
 
-    def test_suspended_job_node(self):
-        "Test that a suspended job goes back to the same node on re queue"
 
-        a = {'resources_available.ncpus': 1}
-        self.server.create_vnodes('vnode', a, 2, self.mom, usenatvnode=True)
+
+    def test_held_jobs_equiv_class(self):
+        """
+        Test that held jobs do not go into another equivalence class.
+        """
+
+        a = {'resources_available.ncpus': 2}
+        self.server.create_vnodes('vnode', a, 1, self.mom, usenatvnode=True)
 
         a = {'Resource_List.ncpus': 1}
         J = Job(TEST_USER, attrs=a)
-        jid1 = self.server.submit(J)
 
-        # create express queue
-        a = {'queue_type': 'e', 'started': 't',
-             'enabled': 't', 'priority': 150}
-        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='expressq')
+        (jid1, jid2, jid3) = self.submit_jobs(3)
 
-        a = {'Resource_List.ncpus': 1, 'queue': 'expressq'}
-        Je = Job(TEST_USER, attrs=a)
-        jid2 = self.server.submit(Je)
 
         self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
-        time.sleep(2)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid3)
 
-        self.scheduler.log_match("Number of job equivalence classes: 1",
-                                 max_attempts=10, starttime=self.t)
-
-        a = {'Resource_List.ncpus': 1, 'queue': 'expressq'}
-        Je = Job(TEST_USER, attrs=a)
-        jid3 = self.server.submit(Je)
-
-        self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
+        self.server.holdjob(jid1)
+        self.server.rerunjob(jobid=jid1, runas=ROOT_USER)
+        self.server.expect(JOB, {'job_state': 'H'}, id=jid1)
 
-        # 1 equi class for j1d, 1 for jid2 and jid3
-        self.scheduler.log_match("Number of job equivalence classes: 2",
-                                 max_attempts=10, starttime=self.t)
-
-        # Delete jid2 and check that jid1 is still in suspended state
-        self.server.deljob(jid2, wait=True)
-        self.server.expect(JOB, {'job_state': 'S'}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
 
-        # delete jid3 and check that jid1 goes back to R state
-        self.server.deljob(jid3, wait=True)
-        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
         self.scheduler.log_match("Number of job equivalence classes: 1",
                                  max_attempts=10, starttime=self.t)
```

Comment:
reset time self.t after you are holding the job to be sure it is not the first one.

Reply:
reset time in tests where log_match is called multiple times wherever necessary